synfig-core/src/modules/mod_svg/svg_parser.cpp

```diff
@@ -2113,3 +2063,173 @@ get_inkscape_version(const xmlpp::Element* svgNodeElement)
+		"font-family", "font-size", "font-size-adjust", "font-stretch", "font-style", "font-variant", "font-weight",
+		"glyph-orientation-horizontal", "glyph-orientation-vertical",
+		"image-rendering", "letter-spacing", "lighting-color",
+		"marker-end", "marker-mid", "marker-start", "mask", "opacity", "overflow", "paint-order", "pointer-events",
+		"shape-rendering", "stop-color", "stop-opacity",
+		"stroke", "stroke-dasharray", "stroke-dashoffset", "stroke-linecap", "stroke-linejoin", "stroke-miterlimit", "stroke-opacity", "stroke-width",
+		"text-anchor", "text-decoration", "text-overflow", "text-rendering", "unicode-bidi",
+		"vector-effect", "visibility", "white-space", "word-spacing", "writing-mode"};
+
+	for (const auto& prop : general_props)
+		push(prop, elem->get_attribute_value(prop));
+
+	const std::string elem_name = elem->get_name();
+	if (elem_name == "circle" || elem_name == "ellipse") {
+		for (const auto& prop : circle_ellipse_props)
+			push(prop, elem->get_attribute_value(prop));
+	}
+	if (elem_name == "foreignObject" || elem_name == "image" || elem_name == "rect" || elem_name == "svg" || elem_name == "symbol" || elem_name == "use") {
+		for (const auto& prop : geometry_props)
+			push(prop, elem->get_attribute_value(prop));
+	}
+	if (elem_name == "circle") {
+		push("r", elem->get_attribute_value("r"));
+	}
+	if (elem_name == "circle" || elem_name == "rect") {
+		for (const auto& prop : circle_rect_props)
+			push(prop, elem->get_attribute_value(prop));
+	}
+	if (elem_name == "path") {
+		push("d", elem->get_attribute_value("d"));
+	}
+	// 'fill' for all non-animation elements
+	if (elem_name != "animate" && elem_name != "animateMotion" && elem_name != "animateTransform" && elem_name != "discard" && elem_name != "set") {
+		push("fill", elem->get_attribute_value("fill"));
+	}
+	// transform
+	if (elem_name == "pattern") {
+		push("transform", elem->get_attribute_value("patternTransform"));
+	} else if (elem_name == "linearGradient" || elem_name == "radialGradient") {
+		push("transform", elem->get_attribute_value("gradientTransform"));
+	} else {
+		push("transform", elem->get_attribute_value("transform"));
+	}
+}
+
+void
+Style::merge_style_string(const std::string &style_str)
+{
+	size_t previous_pos = 0;
+	size_t pos = 0;
+	while ((pos = style_str.find(';', pos)) != std::string::npos) {
+		std::string token = style_str.substr(previous_pos, pos-previous_pos);
+		size_t separator_pos = token.find(':');
+		if (separator_pos != std::string::npos && separator_pos != token.size()-1) {
+			std::string prop = synfig::trim(token.substr(0, separator_pos));
+			std::string value = synfig::trim(token.substr(separator_pos+1));
+			if (!prop.empty() && !value.empty())
+				push(prop, value);
+		}
+		previous_pos = pos;
+		pos++;
+	}
+}
+
+void
+Style::push(const std::string &property, const std::string &value)
+{
+	if (!value.empty() && value != "inherit" && !property.empty())
```

Comment:
Am I correct and `inherit` is just ignored now?

Reply:
Yes.
Currently it is too.

In most cases, it is the natural way of CSS style processing. And the current code and my proposal use it by default if not overridden.

My proposal to a Style object have all explicitly defined properties. When you go to a child element, you take the parent style and merge the child properties. 'inherit' would mean 'keep that value from my parent'.

Some properties, however, are non-inherited by default, and `inherit` would force it. https://developer.mozilla.org/en-US/docs/Web/CSS/inheritance#non-inherited_properties I'll inspect later (I hope) if any of SVG needs it.